Initial commit for kubevirt chart

cynepco3hahue opened the pull request:
KubeVirt is a virtual machine management add-on for Kubernetes. The aim is to provide a common ground for virtualization solutions on top of Kubernetes.

I reopen #2669 because I had some time to test it on the k8s environment :smile: 
Signed-off-by: Lukianov Artyom <alukiano@redhat.com>

*****************************************************************************************

Thank you for contributing to kubernetes/charts. Before you submit this PR we'd like to
make sure you are aware of our technical requirements and best practices:

* https://github.com/kubernetes/charts/blob/master/CONTRIBUTING.md#technical-requirements
* https://github.com/kubernetes/helm/tree/master/docs/chart_best_practices

For a quick overview across what we will look at reviewing your PR, please read
our review guidelines:

* https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md

Following our best practices right from the start will accelerate the review process and
help get your PR merged quicker.

When updates to your PR are requested, please add new commits and do not squash the 
history. This will make it easier to identify new changes. The PR will be squashed 
anyways when it is merged. Thanks.

*****************************************************************************************

Commit:
Initial commit for kubevirt chart

Signed-off-by: Lukianov Artyom <alukiano@redhat.com>

cynepco3hahue commented:
```
# helm install --name kubevirt kubevirt/kubevirt --set kubevirt.apiserver.externalips={10.35.72.229} --set kubevirt.spiceproxy.externalip=10.35.72.229
NAME:   kubevirt
LAST DEPLOYED: Tue Dec 12 18:09:36 2017
NAMESPACE: default
STATUS: DEPLOYED

RESOURCES:
==> v1beta1/Deployment
NAME             DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
haproxy          1        1        1           0          0s
spice-proxy      1        1        1           0          0s
virt-api         1        1        1           0          0s
virt-controller  2        2        2           0          0s

==> v1/Pod(related)
NAME                              READY  STATUS             RESTARTS  AGE
libvirt-4nvh7                     0/3    ContainerCreating  0         0s
libvirt-958db                     0/3    ContainerCreating  0         0s
libvirt-sgb8g                     0/3    ContainerCreating  0         0s
libvirt-vrglh                     0/3    ContainerCreating  0         0s
virt-handler-9v8nr                0/1    ContainerCreating  0         0s
virt-handler-ncfqv                0/1    ContainerCreating  0         0s
virt-handler-q4vkx                0/1    ContainerCreating  0         0s
virt-handler-zf862                0/1    ContainerCreating  0         0s
haproxy-585d546c7f-r9rxt          0/1    ContainerCreating  0         0s
spice-proxy-84d697b7b5-74ls7      0/1    ContainerCreating  0         0s
virt-api-5f64ccc84-xwnv8          0/1    ContainerCreating  0         0s
virt-controller-6c78648754-6dzbd  0/1    ContainerCreating  0         0s
virt-controller-6c78648754-84f8b  0/1    ContainerCreating  0         0s

==> v1/ServiceAccount
NAME            SECRETS  AGE
kubevirt-admin  1        0s
kubevirt-infra  1        0s

==> v1beta1/CustomResourceDefinition
NAME                                   AGE
migrations.kubevirt.io                 0s
virtualmachinereplicasets.kubevirt.io  0s
virtualmachines.kubevirt.io            0s
==> v1beta1/ClusterRole
kubevirt-infra  0s

==> v1beta1/ClusterRoleBinding
NAME                          AGE
kubevirt-infra                0s
kubevirt-infra-cluster-admin  0s

==> v1/Service
NAME         TYPE       CLUSTER-IP    EXTERNAL-IP   PORT(S)   AGE
haproxy      ClusterIP  10.102.9.134  10.35.72.229  8184/TCP  0s
spice-proxy  ClusterIP  10.97.14.64   10.35.72.229  3128/TCP  0s
virt-api     ClusterIP  10.99.120.47  <none>        8183/TCP  0s

==> v1beta1/DaemonSet
NAME          DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  NODE SELECTOR  AGE
libvirt       4        4        0      4           0          <none>         0s
virt-handler  4        4        0      4           0          <none>         0s


NOTES:
The KubeVirt components have been installed.

The VM API can be accessed by [10.35.72.229]:8184.
In the case you want to create VM via kubectl command, use
kubectl -s [10.35.72.229]:8184 -f vm.yaml


# kubectl get pods --all-namespaces
NAMESPACE     NAME                                                              READY     STATUS    RESTARTS   AGE
default       spice-proxy-84d697b7b5-74ls7                                      1/1       Running   0          3m
kube-system   etcd-kubernetes-master.qa.lab.tlv.redhat.com                      1/1       Running   1          23m
kube-system   haproxy-585d546c7f-r9rxt                                          1/1       Running   0          3m
kube-system   kube-apiserver-kubernetes-master.qa.lab.tlv.redhat.com            1/1       Running   1          23m
kube-system   kube-controller-manager-kubernetes-master.qa.lab.tlv.redhat.com   1/1       Running   1          23m
kube-system   kube-dns-545bc4bfd4-jfqgq                                         3/3       Running   0          1h
kube-system   kube-proxy-bz95x                                                  1/1       Running   0          8h
kube-system   kube-proxy-k5crg                                                  1/1       Running   1          8h
kube-system   kube-proxy-mk5vx                                                  1/1       Running   1          8h
kube-system   kube-proxy-nrksq                                                  1/1       Running   0          8h
kube-system   kube-scheduler-kubernetes-master.qa.lab.tlv.redhat.com            1/1       Running   1          23m
kube-system   libvirt-4nvh7                                                     3/3       Running   0          3m
kube-system   libvirt-958db                                                     3/3       Running   0          3m
kube-system   libvirt-sgb8g                                                     3/3       Running   0          3m
kube-system   libvirt-vrglh                                                     3/3       Running   0          3m
kube-system   tiller-deploy-84b97f465c-v4wpf                                    1/1       Running   0          1h
kube-system   virt-api-5f64ccc84-xwnv8                                          1/1       Running   0          3m
kube-system   virt-controller-6c78648754-6dzbd                                  1/1       Running   0          3m
kube-system   virt-controller-6c78648754-84f8b                                  0/1       Running   0          3m
kube-system   virt-handler-9v8nr                                                1/1       Running   0          3m
kube-system   virt-handler-ncfqv                                                1/1       Running   0          3m
kube-system   virt-handler-q4vkx                                                1/1       Running   0          3m
kube-system   virt-handler-zf862                                                1/1       Running   0          3m
kube-system   weave-net-27qkq                                                   2/2       Running   0          8h
kube-system   weave-net-bnd2p                                                   2/2       Running   2          8h
kube-system   weave-net-ggnmr                                                   2/2       Running   3          8h
kube-system   weave-net-gw7km                                                   2/2       Running   0          8h
```

fabiand commented:
@cynepco3hahue did you take the `kubevirt.yaml` file form the 0.1.0 release?

I wonder, because the chart contains multiple individual files - and I do not see the single kubevirt.yaml.

cynepco3hahue commented:
@fabiand Yes, I adapted manifests from `kubevirt.yaml`. I just saw that under chart common practice to separate all manifests because I think it will make things easier if you want to edit single entity

fabiand commented:
Right - The thing is that this is more manual work whenever there is a update from KubeVirt.

I'd suggest to take the kubevirt.yaml as is, and have separate manifests for whatever else is needed.

cynepco3hahue commented:
@fabiand The question is if a `chart` guys will accept it.

cynepco3hahue commented:
/assign @lachie83

Commit:
Allign manifests with the last KubeVirt release

Signed-off-by: Lukianov Artyom <alukiano@redhat.com>

cynepco3hahue commented:
Verification
```
# helm install --name kubevirt kubevirt/kubevirt
NAME:   kubevirt
LAST DEPLOYED: Tue Jan  9 11:18:23 2018
NAMESPACE: default
STATUS: DEPLOYED

RESOURCES:
==> v1/ServiceAccount
NAME            SECRETS  AGE
kubevirt-admin  1        1s
kubevirt-infra  1        1s

==> v1beta1/CustomResourceDefinition
NAME                                   AGE
migrations.kubevirt.io                 1s
virtualmachinereplicasets.kubevirt.io  1s
virtualmachines.kubevirt.io            1s

==> v1beta1/ClusterRole
kubevirt-infra  1s

==> v1beta1/ClusterRoleBinding
NAME                          AGE
kubevirt-infra                1s
kubevirt-infra-cluster-admin  1s

==> v1beta1/DaemonSet
NAME          DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  NODE SELECTOR  AGE
libvirt       4        4        0      4           0          <none>         1s
virt-handler  4        4        0      4           0          <none>         1s

==> v1beta1/Deployment
NAME             DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
virt-controller  2        2        2           0          1s

==> v1/Pod(related)
NAME                             READY  STATUS             RESTARTS  AGE
libvirt-4jhxn                    0/3    ContainerCreating  0         1s
libvirt-fggrt                    0/3    ContainerCreating  0         1s
libvirt-jvz7n                    0/3    ContainerCreating  0         1s
libvirt-mhtmc                    0/3    ContainerCreating  0         1s
virt-handler-5w5gm               0/1    ContainerCreating  0         1s
virt-handler-bjppt               0/1    ContainerCreating  0         1s
virt-handler-c4zt7               0/1    ContainerCreating  0         1s
virt-handler-vggjr               0/1    ContainerCreating  0         1s
virt-controller-f7966ccdf-ph89d  0/1    ContainerCreating  0         1s
virt-controller-f7966ccdf-t75x7  0/1    ContainerCreating  0         1s


NOTES:
The KubeVirt components have been installed.

The VM API can be accessed via kubectl.
For example to get all vms you can run:
# kubectl get vms

[root@kubernetes-master ~]# kubectl get pods
No resources found, use --show-all to see completed objects.
[root@kubernetes-master ~]# kubectl get pods -n kube-system
NAME                                                              READY     STATUS    RESTARTS   AGE
etcd-kubernetes-master.qa.lab.tlv.redhat.com                      1/1       Running   2          27d
kube-apiserver-kubernetes-master.qa.lab.tlv.redhat.com            1/1       Running   2          27d
kube-controller-manager-kubernetes-master.qa.lab.tlv.redhat.com   1/1       Running   2          27d
kube-dns-545bc4bfd4-jfqgq                                         3/3       Running   0          27d
kube-proxy-bz95x                                                  1/1       Running   1          28d
kube-proxy-k5crg                                                  1/1       Running   0          28d
kube-proxy-mk5vx                                                  1/1       Running   0          28d
kube-proxy-nrksq                                                  1/1       Running   1          28d
kube-scheduler-kubernetes-master.qa.lab.tlv.redhat.com            1/1       Running   2          27d
libvirt-4jhxn                                                     3/3       Running   0          1m
libvirt-fggrt                                                     3/3       Running   0          1m
libvirt-jvz7n                                                     3/3       Running   0          1m
libvirt-mhtmc                                                     3/3       Running   0          1m
tiller-deploy-84b97f465c-v4wpf                                    1/1       Running   1          27d
virt-controller-f7966ccdf-ph89d                                   0/1       Running   0          1m
virt-controller-f7966ccdf-t75x7                                   1/1       Running   0          1m
virt-handler-5w5gm                                                1/1       Running   0          1m
virt-handler-bjppt                                                1/1       Running   0          1m
virt-handler-c4zt7                                                1/1       Running   0          1m
virt-handler-vggjr                                                1/1       Running   0          1m
weave-net-27qkq                                                   2/2       Running   3          28d
weave-net-bnd2p                                                   2/2       Running   0          28d
weave-net-ggnmr                                                   2/2       Running   1          28d
weave-net-gw7km                                                   2/2       Running   2          28d
```

cynepco3hahue commented:
@rmohr Without external IP's it looks much better :smile:

fabiand commented on incubator/kubevirt/README.md:
```diff
@@ -0,0 +1,45 @@
+|                Parameter                  |            Description              |          Default           |
+| ----------------------------------------- | ------------------------------------| -------------------------- |
+| `kubevirt.repository`                     |haproxy container name               | `kubevirt`                 |
+| `kubevirt.tag`                            |haproxy container tag                | `latest`                   |
```
The two parameteres are not specific to haproxy, but used for all images

as the chart gives version 0.1.0, the tag should also default to this version.

cynepco3hahue replied:
Missed this one

fabiand commented on incubator/kubevirt/Chart.yaml:
```diff
@@ -0,0 +1,9 @@
+name: kubevirt
+home: https://github.com/kubevirt/kubevirt
+version: 0.1.0
```
Here 0.1.0 is given, but the images are pulled from `latest` by default, this is inconsistent.

I'd suggest to update this to 0.2.0 (which latest is pointing to).

cynepco3hahue replied:
It chart version, it a little strange that I have version `0.2.0` without `0.1.0`, but it can be a good idea to couple it with KubeVirt releases

fabiand commented on incubator/kubevirt/values.yaml:
```diff
@@ -0,0 +1,3 @@
+kubevirt:
+  repository: "kubevirt"
+  tag: "latest"
```
Should be 0.1.0 (or 0.2.0 after the update).

fabiand reviewed (commented):
Just some small things, in general nice.

I'd also recommend to take the kubevirt.yaml as is and not break it up, in order to reduce the maintenance overhead.

cynepco3hahue commented:
Like I said above put all manifests to a single YAML file, can be not accepted by maintainers of helm charts.
I prefer to wait for some output from their side.

fabiand commented:
@sameersbn upstream provides a single reference yaml file which could be used for the helm chart. Do you have objects in using a single yaml file, which includes multiple object definitions, in a helm chart?
Eventually providing additional yaml files to fill some gaps.

cynepco3hahue commented:
@unguiculus Can you please take a look?

Commit:
Update manifest to v0.3.0-alpha.2

Signed-off-by: Lukianov Artyom <alukiano@redhat.com>

cynepco3hahue commented:
Verification
```
# helm install --name kubevirt kubevirt/kubevirt
NAME:   kubevirt
LAST DEPLOYED: Sun Feb 18 09:14:42 2018
NAMESPACE: default
STATUS: DEPLOYED

RESOURCES:
==> v1beta1/ClusterRoleBinding
NAME                               AGE
kubevirt-controller-cluster-admin  0s
kubevirt-controller                0s
kubevirt-privileged-cluster-admin  0s

==> v1beta1/DaemonSet
NAME          DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  NODE SELECTOR  AGE
virt-handler  1        1        0      1           0          <none>         0s

==> v1beta1/Deployment
NAME             DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
virt-controller  2        2        2           0          0s

==> v1/Pod(related)
NAME                              READY  STATUS             RESTARTS  AGE
virt-handler-d2prg                0/1    ContainerCreating  0         0s
virt-controller-7fd9c68494-6twjf  0/1    ContainerCreating  0         0s
virt-controller-7fd9c68494-8m4hb  0/1    ContainerCreating  0         0s

==> v1/ServiceAccount
NAME                 SECRETS  AGE
kubevirt-controller  1        0s
kubevirt-privileged  1        0s

==> v1beta1/CustomResourceDefinition
NAME                                   AGE
virtualmachinereplicasets.kubevirt.io  0s
virtualmachines.kubevirt.io            0s

==> v1beta1/ClusterRole
kubevirt-controller  0s


NOTES:
The KubeVirt components have been installed.

The VM API can be accessed via kubectl.
For example to get all vms you can run:
# kubectl get vms

etcd-master                        1/1       Running   1          8m                                              
kube-apiserver-master              1/1       Running   1          8m
kube-controller-manager-master     1/1       Running   1          9m
kube-dns-6f4fd4bdf-dk5rf           3/3       Running   3          9m
kube-flannel-ds-mjrzt              1/1       Running   2          9m
kube-proxy-hjwqv                   1/1       Running   1          9m
kube-scheduler-master              1/1       Running   1          9m
tiller-deploy-59d854595c-ttjvx     1/1       Running   0          2m
virt-controller-7fd9c68494-6twjf   0/1       Running   0          1m
virt-controller-7fd9c68494-8m4hb   1/1       Running   0          1m
virt-handler-d2prg                 1/1       Running   0          1m
```

jdumars commented:
/ok-to-test

unguiculus commented:
/assign
/hold

unguiculus commented:
Sorry for the delay. Please apply best practices, especially for labels, resource names, RBAC.

https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md
https://github.com/kubernetes/helm/blob/master/docs/chart_best_practices/rbac.md

Please compare with what you get from `helm create` using Helm 2.8+ and apply accordingly.

cynepco3hahue commented:
@unguiculus Thanks for replay, I will update the PR at the beginning of the next week.

Commit:
Apply last changes

- change RBAC manifests to use best practies
- adapt changes to kubevirt version 0.4.1

Signed-off-by: Lukianov Artyom <alukiano@redhat.com>

cynepco3hahue commented:
@unguiculus I updated chart with last changes and tested it on
```
# kubectl version
Client Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T12:22:21Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"9", GitVersion:"v1.9.3", GitCommit:"d2835416544f298c919e2ead3be3d0864b52323b", GitTreeState:"clean", BuildDate:"2018-02-07T11:55:20Z", GoVersion:"go1.9.2", Compiler:"gc", Platform:"linux/amd64"}
```
```bash
# helm install --name kubevirt kubevirt/kubevirt
NAME:   kubevirt
LAST DEPLOYED: Tue Apr 24 08:23:47 2018
NAMESPACE: default
STATUS: DEPLOYED

RESOURCES:
==> v1beta1/DaemonSet
NAME          DESIRED  CURRENT  READY  UP-TO-DATE  AVAILABLE  NODE SELECTOR  AGE
virt-handler  1        1        0      1           0          <none>         1s

==> v1/Pod(related)
NAME                              READY  STATUS             RESTARTS  AGE
virt-handler-ldwh5                0/1    ContainerCreating  0         1s
virt-api-55f58b8865-qkp5c         0/1    ContainerCreating  0         1s
virt-controller-845f96c7c4-5rx4w  0/1    ContainerCreating  0         1s
virt-controller-845f96c7c4-kg26t  0/1    ContainerCreating  0         1s

==> v1/ServiceAccount
NAME                 SECRETS  AGE
kubevirt-apiserver   1        1s
kubevirt-controller  1        1s
kubevirt-privileged  1        1s

==> v1beta1/ClusterRoleBinding
NAME                               AGE
kubevirt-apiserver-auth-delegator  1s
kubevirt-apiserver                 1s
kubevirt-controller-cluster-admin  1s
kubevirt-controller                1s
kubevirt-privileged-cluster-admin  1s

==> v1/Service
NAME             TYPE       CLUSTER-IP     EXTERNAL-IP  PORT(S)   AGE
virt-api         ClusterIP  10.102.12.85   <none>       443/TCP   1s
virt-controller  ClusterIP  10.108.36.212  <none>       8182/TCP  1s

==> v1beta1/RoleBinding
NAME                AGE
kubevirt-apiserver  1s

==> v1beta1/Deployment
NAME             DESIRED  CURRENT  UP-TO-DATE  AVAILABLE  AGE
virt-api         1        1        1           0          1s
virt-controller  2        2        2           0          1s

==> v1beta1/CustomResourceDefinition
NAME                                   AGE
offlinevirtualmachines.kubevirt.io     1s
virtualmachinereplicasets.kubevirt.io  1s
virtualmachines.kubevirt.io            1s
virtualmachinepresets.kubevirt.io      1s

==> v1beta1/ClusterRole
kubevirt-apiserver   1s
kubevirt-controller  1s

==> v1beta1/Role
kubevirt-apiserver  1s


NOTES:
The KubeVirt components have been installed.

The VM API can be accessed via kubectl.
For example to get all vms you can run:
# kubectl get vms

[root@node01 ~]# kubectl -n kube-system get pods
NAME                               READY     STATUS              RESTARTS   AGE
etcd-node01                        1/1       Running             0          13m
kube-apiserver-node01              1/1       Running             0          13m
kube-controller-manager-node01     1/1       Running             0          13m
kube-dns-6f4fd4bdf-sztm8           3/3       Running             0          14m
kube-flannel-ds-l8jjh              1/1       Running             1          14m
kube-proxy-vgd6z                   1/1       Running             0          14m
kube-scheduler-node01              1/1       Running             0          13m
tiller-deploy-865dd6c794-4x7d4     1/1       Running             0          8m
virt-api-55f58b8865-qkp5c          0/1       ContainerCreating   0          9s
virt-controller-845f96c7c4-5rx4w   0/1       ContainerCreating   0          9s
virt-controller-845f96c7c4-kg26t   0/1       ContainerCreating   0          9s
virt-handler-ldwh5                 1/1       Running
```
I used `helm-v2.8.2`

fabiand commented:
That really looks good.

But with my kubevirt glasses on it is quite some significant work to provide and keep the helm chart updated.

I wonder if this can be automated ..

+1 from my side on this PR

cynepco3hahue commented:
@fabiand I thought about it, the question if we can use templates to generate helm template :smile: 
I will check it on my free time

rmohr commented:
@cynepco3hahue how about adding helm to our kubevirt-ci, adding the helm charts there and let our kubevirt-bot do pull requests on this repo if things change for a new release and our tests with the new helm chart pass?

@unguiculus would you mind an automated process from our side? At least once we managed to take the first hurdle of being added at all ...

unguiculus commented:
@rmohr @cynepco3hahue We wouldn't mind an automated process. Just be aware that folks will create PRs over here as well.

Please checkout our guidelines again. Make sure resource names are based on the fullname partial (exception of CRDs, of course). Also, make sure you use the latest `_helpers.tpl`.

fabiand commented:
Cool.

cynepco3hahue commented:
I did another round on https://github.com/kubernetes/charts/blob/master/REVIEW_GUIDELINES.md, and I hope I did not miss anything.
@unguiculus Can you please review it again?

cynepco3hahue commented:
/retest

unguiculus commented on incubator/kubevirt/templates/apiserver-deployment.yaml:
```diff
@@ -0,0 +1,37 @@
+        - name: virt-api
+          image: "{{ .Values.kubevirt.repository }}/virt-api:{{ .Values.kubevirt.tag }}"
+          imagePullPolicy: {{ .Values.kubevirt.pullPolicy }}
+          command:
```
Indent two spaces.

cynepco3hahue replied:
fixed

unguiculus commented on incubator/kubevirt/values.yaml:
```diff
@@ -0,0 +1,20 @@
+  # Specifies whether a service accounts should be created
+  apiserver:
+    create: true
+    name: "kubevirt-apiserver"
```
Leave service account names empty by default.

cynepco3hahue replied:
fixed

unguiculus commented on incubator/kubevirt/Chart.yaml:
```diff
@@ -0,0 +1,12 @@
+keywords:
+- virtualization
+maintainers:
+- name: Artyom Lukianov
```
Use Github username.

cynepco3hahue replied:
fixed

unguiculus commented on incubator/kubevirt/templates/apiserver-deployment.yaml:
```diff
@@ -0,0 +1,37 @@
+apiVersion: extensions/v1beta1
```
You should use `apps/vbeta2` for deployments and daemonsets.

cynepco3hahue replied:
fixed

unguiculus reviewed (changes requested):
Is the namespace `kube-system` required? Otherwise I'd let the user decide. Generally, installing stuff into `kube-system` is an anti-pattern.

fabiand commented:
Well - To me it look slike it's a convention to add true cluster enhanving namespace aware add-ons to `kube-system`.

On that side I'd actually love to see that wg-app-def can formalize true add-ons - their metadata, deployment, …

rmohr commented:
To avoid misunderstandings: Like @fabiand pointed out, this is a cluster-add-on. While it is not entirely clear to me on what exactly is still recommended from the [critical-add-on-documentation](https://kubernetes.io/docs/tasks/administer-cluster/guaranteed-scheduling-critical-addon-pods/), `kube-system` is mentioned there as the place to go. Further the `user` in the context of the helm-chart is an administrator.

@unguiculus but we can of course make it configurable. Are there other examples for such add-ons which are already accepted?

Commit:
Review comments fixes

Signed-off-by: Lukianov Artyom <alukiano@redhat.com>

cynepco3hahue commented:
@unguiculus Hi when I set API version `apps/vbeta2` for deployments and daemon sets I can see CI is failing on
```
Error: apiVersion "apps/vbeta2" in kubevirt/templates/virthandler-daemonset.yaml is not available
```

unguiculus commented:
Sorry, that was a typo. Should be `apps/v1beta2`.

Commit:
Add selector to DaemonSet and Deployments

Signed-off-by: Lukianov Artyom <alukiano@redhat.com>

unguiculus commented on incubator/kubevirt/templates/apiserver-deployment.yaml:
```diff
@@ -0,0 +1,37 @@
+  labels:
+{{ include "labels.standard" . | indent 4 }}
+    component: "virt-api-server"
+spec:
```
You need a `spec.selector`

cynepco3hahue replied:
fixed

unguiculus commented on incubator/kubevirt/templates/apiserver-service.yaml:
```diff
@@ -0,0 +1,16 @@
+      targetPort: 8443
+      protocol: TCP
+  selector:
+{{ include "selectors.standard" . | indent 4 }}
```
`chart` and `heritage` should not be part of the selector.

cynepco3hahue replied:
fixed

unguiculus commented on incubator/kubevirt/templates/controller-deployment.yaml:
```diff
@@ -0,0 +1,50 @@
+spec:
+  selector:
+    matchLabels:
+{{ include "labels.standard" . | indent 6 }}
```
`chart` and `heritage` should not be part of the selector.

unguiculus commented on incubator/kubevirt/templates/controller-service.yaml:
```diff
@@ -0,0 +1,15 @@
+    - port: 8182
+      targetPort: virt-controller
+  selector:
+{{ include "selectors.standard" . | indent 4 }}
```
`chart` and `heritage` should not be part of the selector.

cynepco3hahue replied:
fixed

unguiculus commented on incubator/kubevirt/values.yaml:
```diff
@@ -0,0 +1,21 @@
+  repository: "kubevirt"
+  tag: "v0.4.1"
+  pullPolicy: IfNotPresent
+  namespace: "kube-system"
```
Remove the namespace here and from all resources. It can be specified upon installation and does not need any special configuration.

cynepco3hahue replied:
done

cynepco3hahue replied:
@unguiculus I can see that tests now are failing on
```
Error: release kubevir-10072-1 failed: ClusterRoleBinding.rbac.authorization.k8s.io "kubevir-10072-1-kubevirt-apiserver-auth-delegator" is invalid: subjects[0].namespace: Required value
```
Can you please specify how can I pass namespace upon installation?

unguiculus commented on incubator/kubevirt/values.yaml:
```diff
@@ -0,0 +1,21 @@
+kubevirt:
```
Images should be configure like so:

```yaml
image:
  repository:
  tag:
  pullPolicy: 
```

cynepco3hahue replied:
done

Commit:
Remove namespace from resources

Signed-off-by: Lukianov Artyom <alukiano@redhat.com>
